### ISSUE TITLE
RData files are 

### ISSUE THREAD

**Issue by jamesblackshaw** [hfang-bristol/RDataCentre#1]
I am using R3.2.3 and a fresh install of XGR.

When installing I got the following error messages:
1: In install.packages(update[instlib == l, "Package"], l, contriburl = contriburl,  :
  installation of package ‘rJava’ had non-zero exit status
2: In install.packages(update[instlib == l, "Package"], l, contriburl = contriburl,  :
  installation of package ‘nlme’ had non-zero exit status
I installed nlme by itself, it loaded fine. 
R java failed, leaving me with RJava 0.9-7 rather than 0.9-8

I have been trying to load the RData files and get the following error message. This happens whether I load the RData object with your load function or with R's default load() function.

"Error: bad restore file magic number (file may be corrupted) -- no data loaded
In addition: Warning message:
file ‘ig.EF.RData’ has magic number 'versi'
  Use of save versions prior to 2 is deprecated"

I can't tell if this is a problem with your files of a requirement to have the latest version of rJava.

**Comment by hfang-bristol** [comment]
Hi jamesblackshaw

*It is very weird!*

*Can you first try the following instructions to install XGR?*

*If it works, you should use this function '*xRDataLoader*' to load the
RData.*

*Hai*


2 Installation

We assume R <http://cran.r-project.org/>, a language and environment for
statistical computing and graphics, has been installed. For installation of
the XGR package, please follow the instructions below:

   - Install the the XGR package and its dependencies:

source("http://bioconductor.org/biocLite.R")biocLite("XGR")


   - Highly recommend to install the development version of XGR package
   (now hosted in *GitHub*2 <http://xgr.r-forge.r-project.org/#fn2>):

if(!("devtools" %in% rownames(installed.packages())))
install.packages("devtools")library(devtools)install_github(c("hfang-bristol/XGR",
"hfang-bristol/dnet"), dependencies=T)

**Comment by jamesblackshaw** [comment]
I have traced this problem to a certain set of your data files.
I can happily load any of the files starting with "org." using either R load or your custom function.
However the problem comes with a lot of your files that are only ~130 bytes in size and do not contain data.

Checking this on your GitHub page, I saw the same problem.
https://github.com/hfang-bristol/RDataCentre/blob/master/Portal/DrugBank.RData
This does not appear to be an .rdata file, but a text file describing one. It reads:
"version https://git-lfs.github.com/spec/v1
oid sha256:c00875c7df98a88b9e89ddce516cd5e71e95b3d26dd5912ede19b2cedb1e5675
size 911"

Here is a list of the affected files:
GWAS_LD.RData
JKscience_expression.RData
JKscience_genotype.RData
JKscience_TS2B.RData
DrugBank.RData
IlluminaHumanHT.RData
IlluminaOmniExpress.RData
ImmunoBase_LD.RData
JKng_mono.RData
JK_nk.RData
JKscience_TS2A.RData
RegulomeDB_SNPs.RData
ig.EF.RData
ig.GOBP.RData
ig.GOMF.RData
ig.HPPA.RData
ig.MP.RData
ImmunoBase.RData
JKnc_neutro.RData
JKng_bcell.RData
JKscience_TS1A.RData
JKscience_TS3A.RData
GWAS2EF.RData
ig.DO.RData
ig.GOCC.RData
ig.HPCM.RData
ig.HPMA.RData
ig.HPMI.RData

I think you may need to push the files to GitHub again.

**Comment by hfang-bristol** [comment]
The reason for that is because I am using git-lfs (https://git-lfs.github.com) to track RData file. And their is a limitation for the file size. 
Always you can use 'xRDataLoader' function to load RData file via (for example):
org.Hs.egHPPA <- xRDataLoader(RData.customised='org.Hs.egHPPA')
DrugBank <- xRDataLoader(RData.customised='DrugBank.RData')

**Comment by jamesblackshaw** [comment]
I am trying this and it is not working. If I use any of your examples above, I get "Error: bad restore file magic number (file may be corrupted) -- no data loaded"

However, if I clone your git repo from https://github.com/hfang-bristol/RDataCentre.git and set the Portal folder as the working directory, I can load HPPA with xRDataLoader using the following code:
org.Hs.egHPPA <- xRDataLoader(RData.customised= "org.Hs.egHPPA.RData", RData.location="./")

The same code for Drugbank fails with the "Bad restore file" error message.
DrugBank <- xRDataLoader(RData.customised= "DrugBank.RData", RData.location="./")

None of the files in my previous list download correctly from github, leaving me only with the file description. 

Can you please explain how I can download all your data files, since git clone doesn't work and I can't get your default command to?

**Comment by jamesblackshaw** [comment]
Right, I think I understand now. I need to install git-lfs in order to be able to use these files. Do I also need it in order for the default XRDataLoader setting to work? because I can't see it mentioned in the documentation for XGR and if so, it would be helpful to include this as a requirement to avoid future issues.

**Comment by hfang-bristol** [comment]
no. I think there is problem for me to host all RData files in git-lfs.

This is another location from which you can download all of them (
http://galahad.well.ox.ac.uk/bigdata).

Thanks

Hai

**Comment by jamesblackshaw** [comment]
Thanks! For what it's worth, it looks like you're over your data quota on GitHub as when I tried to use git-lfs I got this:

This repository is over its data quota. Purchase more data packs to restore access.
Docs: https://help.github.com/articles/purchasing-additional-storage-and-bandwidth-for-a-personal-account/

**Comment by hfang-bristol** [comment]
Thanks. I have been notified for this. And I will stop using git-lfs. Hai
